pkg/spec/v1alpha1/config.go

```diff
@@ -21,10 +21,11 @@ func New() *Config {
 // Config holds the configuration variables for config version v1alpha1
 // ApiVersion and Kind are currently unused, this may change in the future.
 type Config struct {
-	APIVersion string   `json:"apiVersion"`
-	Kind       string   `json:"kind"`
-	Metadata   Metadata `json:"metadata"`
-	Spec       Spec     `json:"spec"`
+	APIVersion string      `json:"apiVersion"`
+	Kind       string      `json:"kind"`
+	Metadata   Metadata    `json:"metadata"`
+	Spec       Spec        `json:"spec"`
+	Data       interface{} `json:"data"`
```

Comment:
I'd like us to think about this a bit more.

Generally we seem to have two ways here:
1. Having `tanka.dev/Environment` as yet another top level field that is recognized and filtered out by Tanka
2. What this PR currently implements, understanding `Environment` effectively as a data holder type

I would not worry too much about compatibility here. If an older Tanka version encounters an empty `spec.json` / no at all, `apply` will inevitably fail, so no harm can be done.

That being said, Option 1 feels more lightweight to me, it requires less changes to your code (you just embed or `import` `spec.json`).

Ultimately both solutions are fine for me, but I would like hear some arguments pro / contra those :)

/cc @malcolmholmes @jdbaldry 
@igorwwwwwwwwwwwwwwwwwwww @mikulas <-- You two were interested in this before

Reply:
I think the final jsonnet should be deterministic about which environment the data belongs to. This doesn't mean that we have to duplicate the data for each environment, with this proposal/PR I actually want to prevent that.

Example for not-duplicating the data:
```jsonnet
{
  main(namespace, cluster):: {
    config+:: {
      namespace: namespace,
      cluster_name: cluster,
    },
  },

  apiVersion: 'tanka.dev/v1alpha1',
  kind: 'Environment',
  metadata: { labels: { cluster_name: 'dev' } },
  spec: { namespace: 'tanka' },
  data: $.main(self.spec.namespace, self.metadata.labels.cluster_name),
}
```

In another branch (build on this branch) I [implement](https://github.com/grafana/tanka/compare/duologic/inline_environment...duologic/multi_inline_envs#diff-e5a9edc90a041a304d2a9ce3c9cd065defeda6af8b2c264fdd004e0cbd7af311R230) filtering out the tanka.dev/Environment object (similar to `Process()`), which makes the code a lot more clear than in this PR (no more weird `switch`).

For importing spec.json can simply wrap, still looks lightweight to me: :innocent: 
```jsonnet
(import 'spec.json) + { data:
  { // your original main.jsonnet }
}
```
I would like to keep backwards compatability as I don't want to refactor our whole internal code base. :wink: Also, I think the current main.jsonnet/spec.json approach is actually very beginner friendly. I'd propose to present inline environment(s) as an advanced use case in the documentation. WDYT?